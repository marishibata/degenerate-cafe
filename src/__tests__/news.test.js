import React from 'react';
import { render } from '@testing-library/react';
import News from '../Components/news';


jest.mock('axios', () => {
  return {
    default: () => {
      return new Promise((resolve, reject) => {
        resolve({
          data: {data: [
            {
              id: "0abcd",
              title: "Lorem ipsum",
              url: "https://messari.io/article/lorem-ipsum"
            },
            {
              id: "1abcd",
              title: "Testing",
              url: "https://messari.io/article/testing"
            }
          ]}
        })})
    
    }
  }
});


describe('news section', () => {
  it('Renders consistently', async () => {
    const { asFragment } = render(<News/>);
    expect(asFragment()).toMatchSnapshot();
  })

  it('Successfully renders news component', async () => {
    const { findByText } = render(<News/>);
    const result = await findByText('News (Sourced From Messari)');
    expect(result).toBeTruthy();
  })


  it('Successfully fetches news articles', async () => {
    const { findByText } = render(<News/>);
    const result = await findByText('Lorem ipsum')
    expect(result).toBeTruthy();
  })
})