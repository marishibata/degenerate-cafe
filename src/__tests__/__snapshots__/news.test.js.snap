// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`news section Renders consistently 1`] = `
<DocumentFragment>
  <div
    class="news-parent"
  >
    <div
      class="title"
    >
      <span>
        News (Sourced From Messari)
      </span>
    </div>
    <div
      class="news-par"
    />
  </div>
</DocumentFragment>
`;
